static/src/javascripts-legacy/projects/common/modules/experiments/ab.js

```diff
@@ -87,9 +87,12 @@ define([
     }
 
     function getActiveTests() {
-        var now = new Date();
+        var today = new Date();
+        // new Date(test.expiry) sets the expiry time to 00:00:00
+        // Using SetHours allows a test to run until the END of the expiry day
+        today.setHours(0,0,0,0);
```

Comment:
Yaaay arrg! Sounds like this code will work perfectly, thanks for explaining it to me!

I'd say is that this inequality is a little non-obvious, because we're comparing two values that both correspond to 24 hours earlier than the threshold we actually care about (ie when the test is supposed to to end!). I wonder if there is any way we could guard against someone else coming up against this code and being confused by it? I thought about:

* a test in [ab.spec.js](https://github.com/guardian/frontend/blob/ec6a89cc8/static/test/javascripts/spec/common/experiments/ab.spec.js) - but producing a good test would probably involve some kind of injectable clock (ie one you could set `now` to be an arbitrary fixed point in time)
* verbose variable naming, eg : `var startOfToday = new Date();` and `var startOfLastDayOfTest = new Date(test.expiry)`

These are just ideas, I don't want to hold you up on getting this important fix out! So :+1: in any case